In [2]:
# import the libraries needed

import pandas as pd
import numpy as np
import os
import sys
sys.path.append(os.path.abspath(os.path.join('../scripts')))
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import networkx as nx



In [3]:
# ignore the warnings
import warnings
warnings.filterwarnings('ignore')

In [4]:
#load the data that we are to use
df = pd.read_csv("C:/Users/DESMOND/Airflight_analysis/AirportMapping.csv")
df.head()

City  Country Code
0         Aalborg  Denmark  AAL
1        Aalesund   Norway  AES
2          Aarhus  Denmark  AAR
3  Abbotsford, BC   Canada  YXX
4  Abbotsford, BC   Canada  YXX

In [5]:
df

City      Country Code
0            Aalborg      Denmark  AAL
1           Aalesund       Norway  AES
2             Aarhus      Denmark  AAR
3     Abbotsford, BC       Canada  YXX
4     Abbotsford, BC       Canada  YXX
...              ...          ...  ...
1574        Zanzibar     Tanzania  ZNZ
1575        Zaragoza        Spain  ZAZ
1576       Zhengzhou        China  CGO
1577        Zhoushan        China  HSN
1578          Zurich  Switzerland  ZRH

[1579 rows x 3 columns]

In [18]:
data = pd.read_csv('C:/Users/DESMOND/Airflight_analysis/routes_v3.txt', header = None)
data.head()

0      1    2     3    4     5    6  7    8
0  AL242  AL242  AER  2965  KZN  2990  NaN  0  CR2
1  AL242  AL242  ASF  2966  KZN  2990  NaN  0  CR2
2  AL242  AL242  ASF  2966  MRV  2962  NaN  0  CR2
3  AL242  AL242  CEK  2968  KZN  2990  NaN  0  CR2
4  AL242  AL242  CEK  2968  OVB  4078  NaN  0  CR2

### EDA

In [31]:
df.shape

(1579, 3)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1579 entries, 0 to 1578
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   City     1579 non-null   object
 1   Country  1578 non-null   object
 2   Code     1578 non-null   object
dtypes: object(3)
memory usage: 37.1+ KB


the data we have has got 1579  cities, 1578 countries and codes. 

In [10]:
class OverViewAnalysis:
    
    def __init__(self, df):
        
        self.df = df
    
    
    def read_head(self, top=5):
        return self.df.head(top)
    
    # returning the number of rows columns and column information
    def get_info(self):
        row_count, col_count = self.df.shape
    
        print(f"Number of rows: {row_count}")
        print(f"Number of columns: {col_count}")
        print("================================")

        return (row_count, col_count), self.df.info()
    
    # gets number of distnict values in a given coumn
    def get_count(self, column_name):
        return self.df[column_name].value_counts()
    
    # getting the null count for every column
    def get_null_count(self, column_name):
        print("Null values count")
        print(self.df.isnull().sum())
        return self.df.isnull().sum()
    
    
    

In [11]:
overViewAnal = OverViewAnalysis(df)

In [12]:
overViewAnal.read_head()

City  Country Code
0         Aalborg  Denmark  AAL
1        Aalesund   Norway  AES
2          Aarhus  Denmark  AAR
3  Abbotsford, BC   Canada  YXX
4  Abbotsford, BC   Canada  YXX

In [13]:
overViewAnal.get_info()

Number of rows: 1579
Number of columns: 3
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1579 entries, 0 to 1578
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   City     1579 non-null   object
 1   Country  1578 non-null   object
 2   Code     1578 non-null   object
dtypes: object(3)
memory usage: 37.1+ KB


((1579, 3), None)

Now we have the overview analysis of the data that we are working with. we can see the missing values and we deduce there are none
the data types that we have. 

In [16]:
df.isnull().sum()

City       0
Country    1
Code       1
dtype: int64

We have one missinf data in the code and also in the country

In [19]:
# Flight codes to analyse
flights = ['AL8', 'AL54', 'AL2', 'AL3', 'AL4', 'AL5', 'AL6', 'AL12', 'AL56', 'AL7', 'AL9', 'AL10', 'AL 11', 'AL12', 'AL13', 'AL14', 'AL1', 'AL15', 'AL16', 'AL17',
'AL35', 'AL33', 'AL23', 'AL25', 'AL18', 'AL19', 'AL20', 'AL47', 'AL26', 'AL28', 'AL31', 'AL32', 'AL37', 'AL40']


In [21]:
G = nx.Graph()

In [29]:
# find the rout paths with the first data
city = df['City']
city

0              Aalborg
1             Aalesund
2               Aarhus
3       Abbotsford, BC
4       Abbotsford, BC
             ...      
1574          Zanzibar
1575          Zaragoza
1576         Zhengzhou
1577          Zhoushan
1578            Zurich
Name: City, Length: 1579, dtype: object

In [25]:
# Unique cities in the data
df['City'].unique()

array(['Aalborg', 'Aalesund', 'Aarhus', ..., 'Zhengzhou', 'Zhoushan',
       'Zurich'], dtype=object)

In [33]:
# vertices, airport codes as the vertices
v = city